src/v2/Components/FlashBanner.tsx

```diff
@@ -25,7 +22,22 @@ const messages: Record<FlashMessageKey, string | React.FC> = {
   expired_token: "Link expired. Resend verification email.",
 }
 
-export const FlashBanner: React.FunctionComponent<Props> = ({
+export const FlashBanner: React.FunctionComponent = () => {
+  const [messageCode, setMessageCode] = React.useState(null)
+  const isClient = typeof window !== "undefined"
+
+  React.useEffect(() => {
+    if (isClient) {
+      const query = qs.parse(window.location.search.slice(1))
+      query["flash_message"] && setMessageCode(query["flash_message"])
+    }
+  }, [])
+
+  if (!messageCode) return null
```

Comment:
Maybe move this above the effect to show that we're exiting early and to avoid registering the effect inside of react's execution loop?

Reply:
I'm not sure I follow you here.

Reply:
Meaning, `if (!messageCode) return null` can move up above the `React.useEffect` call. Since `messageCode` is null, and we know that via props, we can immediately check for it right at the top of the component and exit early.